src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2397,8 +2468,30 @@ else if (4 >= bulkScale)
                             tdsWriter.writeByte((byte) 0x04);
                         else
                             tdsWriter.writeByte((byte) 0x05);
-
-                        tdsWriter.writeTime((java.sql.Timestamp) colValue, bulkScale);
+                        if (colValue instanceof String) {
```

Comment:
Need to check for `connection.getSendTemporalDataTypesAsStringForBulkCopy()` here too.

Reply:
Same logic applies here, but these are really good observations / questions.